core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -305,4 +334,76 @@ public void start() {
         }
         logger().info("Docker compose has finished running");
     }
-}
\ No newline at end of file
+}
+
+class LocalDockerCompose implements DockerCompose {
+    private final File composeFile;
+    private final String identifier;
+    private String cmd = "";
+    private Map<String, String> env = new HashMap<>();
+
+    public LocalDockerCompose(File composeFile, String identifier) {
+        this.composeFile = composeFile;
+        this.identifier = identifier;
+    }
+
+    @Override
+    public DockerCompose withCommand(String cmd) {
+        this.cmd = cmd;
+        return this;
+    }
+
+    @Override
+    public DockerCompose withEnv(Map<String, String> env) {
+        this.env = env;
+        return this;
+    }
+
+    @Override
+    public void start() {
+        try {
```

Comment:
Please can you use [CommandLine#executableExists](https://github.com/testcontainers/testcontainers-java/blob/master/core/src/main/java/org/testcontainers/utility/CommandLine.java#L58) to check that docker-compose is available on the path first? That way we can fail with an informative error message